### PR TITLE
Fixing RD2 test failures on regression

### PR DESCRIPTION
**Pull request by rjanjanam**
On regression Rd2 tests most of them are failing as click_object_button from salesforce library is not helping with clicking the button
Also there are few other checks to handle

# Critical Changes

# Changes

# Issues Closed

# Community Ideas Delivered

# New Metadata

# Deleted Metadata

### CODE REVIEW

File: robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot [hunk 1]
```diff
@@ -10,7 +10,7 @@ Suite Setup     Run keywords
 ...             Enable RD2
 ...             Open Test Browser
 ...             Setup Test Data
-Suite Teardown  Delete Records and Close Browser
+#Suite Teardown  Delete Records and Close Browser
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot]: you forgot to uncomment this

File: robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot [hunk 2]
```diff
@@ -42,12 +42,10 @@ Create Fixed Recurring Donation With Monthly Installment
...
 
     Go To Page                              Listing                                   npe03__Recurring_Donation__c
-
-    Click Object Button                     New
-    Wait For Modal                          New                                       Recurring Donation
-    # Reload page is a temporary fix till the developers fix the ui-modal
     Reload Page
-    Wait For Modal                          New                                       Recurring Donation
+    Click Link                              New
+    # Reload page is a temporary fix till the developers fix the ui-modal
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot:47]: I think this comment should be near Reload Page Keyword

File: robot/Cumulus/tests/browser/recurring_donations/zcreate_open_erd_for_contact.robot [hunk 3]
```diff
@@ -33,12 +33,11 @@ ${METHOD}  Credit Card
...
 
     Go To Page                             Listing                                            npe03__Recurring_Donation__c
-    Click Object Button                    New
-    Wait For Modal                         New                                                Recurring Donation
 
-    # Reload page is a temporary fix till the developers fix the ui-modal
     Reload Page
-    Wait For Modal                         New                                                Recurring Donation
+    Click Link                              New
+    # Reload page is a temporary fix till the developers fix the ui-modal
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zcreate_open_erd_for_contact.robot:39]: same comment as above

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot [hunk 4]
```diff
@@ -7,7 +7,7 @@ Suite Setup     Run keywords
 ...             Enable RD2
 ...             Open Test Browser
 ...             Setup Test Data
-Suite Teardown  Delete Records and Close Browser
+#Suite Teardown  Delete Records and Close Browser
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot:10]: uncomment suite teardown

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot [hunk 5]
```diff
@@ -48,15 +48,15 @@ Edit Installment Period For An Enhanced Recurring donation record of type open
 
     Go To Page                              Listing                                   npe03__Recurring_Donation__c
     Current Page Should be                  Listing                                   npe03__Recurring_Donation__c
-    Click Object Button                     New
-    Wait For Modal                          New                                       Recurring Donation
-    # Reload page is a temporary fix till the developers fix the ui-modal
     Reload Page
+    Click Link                              New
+    # Reload page is a temporary fix till the developers fix the ui-modal
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot:53]: same here

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month.robot [hunk 6]
```diff
@@ -45,10 +45,10 @@ Edit Day Of Month For Enhanced Recurring donation record of type open
 
     Go To Page                              Listing                                   npe03__Recurring_Donation__c
     Current Page Should Be                  Listing                                   npe03__Recurring_Donation__c
-    Click Object Button                     New
-    Wait For Modal                          New                                       Recurring Donation
-    # Reload page is a temporary fix till the developers fix the ui-modal
     Reload Page
+    Click Link                              New
+        # Reload page is a temporary fix till the developers fix the ui-modal
```
Comment (skristem) [inline robot/Cumulus/tests/browser/recurring_donations/zedit_erd_day_of_month.robot:50]: same here

**Review by skristem — APPROVED** [review]
added couple of minor comments, otherwise looks good

File: robot/Cumulus/resources/RecurringDonationsPageObject.py [hunk 7]
```diff
@@ -15,6 +15,13 @@
 class RDListingPage(BaseNPSPPage, ListingPage):
     object_name = "npe03__Recurring_Donation__c"
 
+    @capture_screenshot_on_error
+    def wait_for_rd2_modal(self):
+        """Based on the button name (Cancel)  or (Save) on the modal footer, selects and clicks on the respective button"""
+        btnlocator = npsp_lex_locators["button-with-text"].format("Save")
+        self.selenium.scroll_element_into_view(btnlocator)
```
Comment (skristem) [inline robot/Cumulus/resources/RecurringDonationsPageObject.py:22]: I dont think u would need to scroll to the element in order to wait for it to be visible

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot (hunk 5 above)
Comment (rjanjanam) [reply robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot]: Removed the comment

File: robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot (hunk 4 above)
Comment (rjanjanam) [reply robot/Cumulus/tests/browser/recurring_donations/zedit_erd_installment_period.robot]: done

File: robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot (hunk 1 above)
Comment (rjanjanam) [reply robot/Cumulus/tests/browser/recurring_donations/zcreate_fixed_erd_record.robot]: done

**Comment by rponti-sforg** [comment]
@rjanjanam I have pulled the branch down and tested the first RD2 test case:
**zcreate_open_erd_record.robot** - **PASS**

That automation passed.  Should I continue with all RD2 test cases that we have defined in the NPSP Regression Checklist?

**Comment by rjanjanam** [comment]
> @rjanjanam I have pulled the branch down and tested the first RD2 test case:
> **zcreate_open_erd_record.robot** - **PASS**
> 
> That automation passed. Should I continue with all RD2 test cases that we have defined in the NPSP Regression Checklist?

@rponti-sforg yes please, please run all of rd2 either headless or regular browser. Thanks
cci task run robot --org <orgname>  -o suites robot/Cumulus/tests/browser/recurring_donations/ -o include feature:RD2 -o vars BROWSER:headlesschrome

**Comment by rponti-sforg** [comment]
@rjanjanam These are the Enhanced Recurring Donation tests that are **passing** now:
1. zcreate_open_erd_record.robot
2. zedit_open_erd_record.robot
3. zcreate_fixed_erd_record.robot
4. Zcreate_open_erd_for_contact.robot

These are the Enhanced Recurring Donation tests that are still **failing**:
1. zedit_erd_installment_period.robot
2. zpause_closed_rd.robot
3. zedit_day_of_month_verify_next_opportunity.robot
4. zedit_erd_day_of_month_with_manual_opportunity.robot
5. zedit_erd_day_of_month.robot

**Pull request merged**